### PR TITLE
[CUMULUS-2020] Update Elasticsearch mapping

### PR DESCRIPTION
**Pull request by bt6453**
**Summary:** Summary of changes

Addresses [CUMULUS-2020: Search Module: Case Match](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2020)

## Changes

* Update mappings in ES
* Fields that were of type `keyword` are now multi-fields. They are of type `text` with a `raw` field that is of type `keyword`
* Update some ES endpoints to use the `raw` field where appropriate

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [x] Adhoc testing
- [x] Integration tests

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -40,6 +42,8 @@ new `update-granules-cmr-metadata-file-links` task.
 - **[1956](https://github.com/nasa/cumulus/issues/1956)**
   - Update the name of the `cumulus_message_adapter_lambda_layer_arn` output from the `cumulus-message-adapter` module to `cumulus_message_adapter_lambda_layer_version_arn`. The output value has changed from being the ARN of the Lambda layer **without a version** to the ARN of the Lambda layer **with a version**.
   - Update the variable name in the `cumulus` and `ingest` modules from `cumulus_message_adapter_lambda_layer_arn` to `cumulus_message_adapter_lambda_layer_version_arn`
+- **CUMULUS-2020**
+  - Elasticsearch data mappings have been updated to improve search. For example, case insensitive searching will now work (e.g. 'MOD' and 'mod' will return the same granule results). To use the improved Elasticsearch queries, [reindex](https://nasa.github.io/cumulus-api/#reindex) to create a new index with the correct types. Then preform a [change index](https://nasa.github.io/cumulus-api/#change-index) operation to use the new index.
```
Comment (Jkovarik) [inline CHANGELOG.md:46]: Should the last two sentences here be in the 'migration instructions' section?

File: CHANGELOG.md [hunk 2]
```diff
@@ -31,6 +31,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - **CUMULUS-2201**
   - Update `dbIndexer` lambda to process requests in serial
   - Fixes ingestPdrWithNodeNameSpec parsePdr provider error
+- **CUMULUS-2020**
+  - Improved Elasticsearch mappings to improve search
```
Comment (laurenfrederick) [inline CHANGELOG.md:35]:
```suggestion
  - Updated Elasticsearch mappings to support case-insensitive search
```

File: CHANGELOG.md (hunk 1 above)
Comment (laurenfrederick) [reply CHANGELOG.md:46]: Yes please

Comment (laurenfrederick) [reply CHANGELOG.md:46]: It can go in both

File: packages/api/es/textFields.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
+ * For certain Elasticsearch aggregations, text fields are more efficient to use
+ * the keyword version, which is stored in [fieldname].raw. This function
+ * converts that field to use the keyword version if its a text field.
+ *
+ * @param {string/number} field - elasticsearch field
+ * @returns {string} - field converted to the right subfield if eligible
+ */
+function convertTextField(field) {
```
Comment (laurenfrederick) [inline packages/api/es/textFields.js:37]: Can you please write a unit test for this function?

**Review by laurenfrederick — CHANGES_REQUESTED** [review]
Did a bunch of testing for this:

- Before reindex tested that adding providers, collections, granules resulted in no data loss
- Reindexed
- Tested adding new provider, collection, granule
- Tested case-insensitive search
- Tested filtering

Left some small comments. Nice work, Brian!

File: CHANGELOG.md (hunk 1 above)
Comment (bt6453) [reply CHANGELOG.md]: Woops missed that part. Updated!

File: CHANGELOG.md (hunk 2 above)
Comment (bt6453) [reply CHANGELOG.md]: Updated!

File: packages/api/es/textFields.js (hunk 3 above)
Comment (bt6453) [reply packages/api/es/textFields.js:37]: Done!

**Review by laurenfrederick — APPROVED** [review]

**Pull request merged**